lib/krb5/plugin.c

```diff
@@ -651,11 +646,6 @@ search_modules(heim_object_t key, heim_object_t value, void *ctx)
 					  s->caller,
 					  path,
 					  p->dsohandle);
-	if (plugins == NULL) {
-	    /* fallback to loading structure directly */
-	    plugins = add_dso_plugin_struct(s->context, path,
```

Comment:
is there a reason you moved this, is it because you don't regard the two mechanisms as peers, rather a fallback?